File: cpp/PetShop.cpp
#include <iostream>
#include <iomanip>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

class PetShop {
private:
    int id;
    string nama_produk;
    double harga_produk;
    int stok_produk;

public:
    PetShop(int id, const string& nama, double harga, int stok) {
        this->id = id;
        this->nama_produk = nama;
        this->harga_produk = harga;
        this->stok_produk = stok;
    }

    int getId() const { return this->id; }
    string getNama() const { return this->nama_produk; }
    double getHarga() const { return this->harga_produk; }
    int getStok() const { return this->stok_produk; }

    static vector<string> getColumnHeaders() {
        return {"ID", "NAMA PRODUK", "HARGA", "STOK"};
    }

    vector<string> getData() const {
        return {
            to_string(this->id),
            this->nama_produk,
            "Rp" + to_string(this->harga_produk),
            to_string(this->stok_produk)
        };
    }
};

File: python/PetShop.py
class PetShop:
    def __init__(self, id, nama, harga, stok):
        self.id = id
        self.nama_produk = nama
        self.harga_produk = harga
        self.stok_produk = stok

    def get_id(self):
        return self.id

    def get_nama(self):
        return self.nama_produk

    def get_harga(self):
        return self.harga_produk

    def get_stok(self):
        return self.stok_produk

    @staticmethod
    def get_column_headers():
        return ["ID", "NAMA PRODUK", "HARGA", "STOK"]

    def get_data(self):
        return [
            str(self.id),
            self.nama_produk,
            f"Rp{self.harga_produk}",
            str(self.stok_produk)
        ]